packages/expo-cli/src/commands/utils/bundledNativeModules.ts

```diff
@@ -0,0 +1,66 @@
+import JsonFile from '@expo/json-file';
+import resolveFrom from 'resolve-from';
+import { ApiV2 } from 'xdl';
+
+import CommandError from '../../CommandError';
+import Log from '../../log';
+
+interface NativeModule {
+  npmPackage: string;
+  versionRange: string;
+}
+type BundledNativeModuleList = NativeModule[];
+export type BundledNativeModules = Record<string, string>;
+
+export async function getBundledNativeModulesAsync(
+  projectRoot: string,
+  sdkVersion: string
+): Promise<BundledNativeModules> {
+  try {
+    return await getBundledNativeModulesFromApiAsync(sdkVersion);
+  } catch {
+    Log.warn(
+      `There seem to be a transient problem with Expo servers, using the cached dependency map (${Log.chalk.bold(
+        'bundledNativeModules.json'
+      )}) from the package "${Log.chalk.bold`expo`}" installed in your project.`
+    );
+    return await getBundledNativeModulesFromExpoPackageAsync(projectRoot);
+  }
+}
+
+async function getBundledNativeModulesFromApiAsync(
+  sdkVersion: string
+): Promise<BundledNativeModules> {
+  const client = ApiV2.clientForUser();
+  const list = await client.getAsync(`sdks/${sdkVersion}/native-modules`);
```

Comment:
Could we maybe add a comment with a link providing more info about this endpoint and what should be returned, I can see this quickly becoming difficult to modify without proper visibility.